src/lib/pickles/inductive_rule.ml

```diff
@@ -64,19 +70,25 @@ type ( 'prev_vars
      , 'a_var
      , 'a_value
      , 'ret_var
-     , 'ret_value )
+     , 'ret_value
+     , 'auxiliary_var
+     , 'auxiliary_value )
      t =
   { identifier : string
   ; prevs : ('prev_vars, 'prev_values, 'widths, 'heights) H4.T(Tag).t
   ; main :
-      'prev_vars H1.T(Id).t -> 'a_var -> 'prev_vars H1.T(E01(B)).t * 'ret_var
+         'prev_vars H1.T(Id).t
+      -> 'a_var
+      -> 'prev_vars H1.T(E01(B)).t * 'ret_var * 'auxiliary_var
```

Comment:
nit: Would be earlier to read with a more expressive, less low level name for this type.